### PR TITLE
Add `FluxPointsDataset.read`

### PR DESCRIPTION
**Pull request by AtreyeeS**
Directly allows to create a `FluxPointsDataset` as proposed in #3747 
This makes the API homogeneous with `MapDataset`

Also, changing `FluxPointsDataset.write` default `overwrite` to `False` to be consistent with `MapDataset.write`

### CODE REVIEW

File: gammapy/datasets/flux_points.py [hunk 1]
```diff
@@ -146,6 +141,32 @@ def write(self, filename, overwrite=True, **kwargs):
...
+
+        Returns
+        -------
+        dataset : `FluxPointsDataset`
+            FluxPointsDataset
+        """
+        from gammapy.estimators import FluxPoints
+
+        filename = make_path(filename)
+        table = Table.read(filename)
```
Comment (adonath) [inline gammapy/datasets/flux_points.py:165]: Is there any specific reason to go though `Table.read()` and not `FluxPoints.read()`?

Comment (AtreyeeS) [reply gammapy/datasets/flux_points.py:165]: `.write` is writing a `Table`, so I did similarly here. Would `FluxPoints.read()` work in all cases?

**Review by registerrier — COMMENTED** [review]
Thanks @AtreyeeS 

I don't know what should be correct behavior here. 
When a `FluxPointsDataset` is read from disk, do we want to re-serialize it? If yes, should we let the content change? Because this is what should happen because of the `FluxPoints` creation.  

I think we are hitting the same issue as discussed in #3807.

**Comment by AtreyeeS** [comment]
It should behave similarly for `Datasets.read()` and `FluxPointsDataset.read()` (without the models serialisation), no?

**Comment by AtreyeeS** [comment]
Adapted to read the masks as discussed in the dev call

**Review by adonath — APPROVED** [review]
Thanks @AtreyeeS, no further comments from my side...

**Review by registerrier — APPROVED** [review]
Thanks @AtreyeeS !

**Pull request merged**